Simperium/src/main/java/com/simperium/android/CredentialsActivity.java

```diff
@@ -376,31 +381,54 @@ public void onClick(DialogInterface dialog, int which) {
             .show();
     }
 
+    private void showDialogErrorLoginReset() {
+        hideDialogProgress();
+        Context context = new ContextThemeWrapper(CredentialsActivity.this, getTheme());
+        new AlertDialog.Builder(context)
+            .setTitle(R.string.simperium_dialog_title_error)
+            .setMessage(getString(R.string.simperium_dialog_message_login_reset, PASSWORD_LENGTH_MINIMUM))
+            .setNegativeButton(android.R.string.cancel, null)
+            .setPositiveButton(R.string.simperium_button_login_reset,
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Uri uri = Uri.parse(getString(R.string.simperium_dialog_button_reset_url, getEditTextString(mInputEmail)));
+                        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+                        startActivity(intent);
+                    }
+                }
+            )
+            .show();
+    }
+
     private void startLogin() {
         final String email = getEditTextString(mInputEmail).trim();
         final String password = getEditTextString(mInputPassword).trim();
 
-        if (isValidPassword(password)) {
+        // Use false in isValidPasswordLength() to check if password meets new minimum length requirement.
+        if (!isValidPasswordLength(false) || isEmailPasswordMatch()) {
```

Comment:
> We aren't doing the regex check here for newlines and tabs, so that means that although newlines or tabs cannot be used when signing up, we will not force someone with them in their password to reset their password, right?

You're right.  The current code does not force them to reset their password if `isValidPassword` returns `false`, but we should.  I updated that `if`/`else` statement in ee17a06.

This is the first platform to enforce the new password policy.  That's why you're seeing some odd behavior with the Simplenote web app.  The web and others will be updated soon.